recipes/SLURP/direct/train_with_wav2vec2.py

```diff
@@ -55,10 +56,17 @@ def compute_forward(self, batch, stage):
         ):
             return p_seq, wav_lens
         else:
-            p_tokens, scores = self.hparams.beam_searcher(
-                wav2vec2_out, wav_lens
+            topk_tokens, topk_lens, _, _ = self.hparams.beam_searcher(
+                wav2vec2_out.detach(), wav_lens
             )
-            return p_seq, wav_lens, p_tokens
+
+            # Select the best hypothesis
+            best_hyps, best_lens = topk_tokens[:, 0, :], topk_lens[:, 0]
+
+            # Convert best hypothesis to list
+            hyps = undo_padding(best_hyps, best_lens)
```

Comment:
this undo_padding also is mysterious. I see it everywhere, but I don't like that we change something that most users already are very used to .

Reply:
fixed